Clear/Utilities.lean

```diff
@@ -82,4 +82,48 @@ lemma evm_eq_symm_of_isPure_ok_ok {evm evm'} {vs vs'} (h : isPure (Ok evm vs) (O
   symm
   aesop_spec
 
+def preservesEvm (s₀ : State) (s₁ : State) : Prop :=
```

Comment:
I think this predicate should also compare evm states in the case where the second state is a `Checkpoint` no? :slightly_smiling_face: